lib/server/controllers/gRPC/GRPCQcFlagController.js

```diff
@@ -26,13 +27,22 @@ class GRPCQcFlagController {
 
     // eslint-disable-next-line require-jsdoc
     async CreateForDataPass({ runNumber, detectorName, passName, flags }) {
+        const lhcPeriods = await LhcPeriodRepository.findOne({ include: [{ association: 'runs', where: { runNumber } }] });
```

Comment:
NEVER use repositories in controller layer. You must have a function from the service for that.
What I would do though is to fetch the run and include the period from it?
You should have a service function `getFullDataPassNameUsingRunPeriod(partialPassName, runNumber)`

Reply:
applied